protocol/lib/eth/util.go

```diff
@@ -21,27 +21,16 @@ const (
 	MaxAddrLen = 32
 )
 
-// ABI Singleton and mutex.
-var bridgeEventAbi *ethabi.ABI
-var once sync.Once
-
-// getBridgeEventAbi returns the ABI (application binary interface) for the Bridge contract.
-func GetBridgeEventAbi() *ethabi.ABI {
-	// Initialize the singleton in a thread-safe way.
-	once.Do(func() {
-		bridgeAbi, err := ethabi.JSON(strings.NewReader(constants.BridgeEventABI))
+// GetBridgeEventAbi returns the ABI (application binary interface) for the Bridge contract.
+func GetBridgeEventAbi() ethabi.ABI {
```

Comment:
I would use a pointer type since the struct seems like it could be pretty large so not pay the copy tax on each invocation of this method.

The caveat is that it isn't truly safe as a pointer type since `UnmarhsalJSON` mutates it so returning a copy would make each instance truly thread safe if that was important otherwise it is only thread safe as long as nobody else calls `UnmarshalJSON` on it.